app.py

```diff
@@ -413,9 +440,6 @@ def prepare_generic_launch(self, engine_name, context, app_path, app_args):
         except Exception:
             self.log_exception("Error executing engine bootstrap script.")
             raise TankError("Error executing bootstrap script. Please see log for details.")
-        finally:
```

Comment:
My reasoning was that we were putting the new code in place because other methods were quite messy about updating the environment so there was no point pretending we'd be trying to keep it clean.